src/generateRequestQRCode.ts

```diff
@@ -0,0 +1,167 @@
+import {Options, RequestData, ErrorCorrectionLevel} from './types'
+import {BloomLogo} from './BloomLogo'
+
+const QRCodeImpl = require('qr.js/lib/QRCode')
+
+const defaultOptions: Options = {
+  hideLogo: false,
+  ecLevel: 'L',
+  size: 128,
+  bgColor: '#fff',
+  fgColor: '#6067f1',
+}
+
+type CellInfo = {
+  top: number
+  left: number
+  size: number
+  fillStyle: string
+}
+
+const makeDot = (ctx: CanvasRenderingContext2D, info: CellInfo) => {
+  const centerX = info.left + info.size / 2
+  const centerY = info.top + info.size / 2
+  const radius = (info.size / 2) * 0.85
+
+  ctx.beginPath()
+  ctx.arc(centerX, centerY, radius, 0, 2 * Math.PI, false)
+  ctx.fillStyle = info.fillStyle
+
+  ctx.fill()
+  ctx.closePath()
+}
+
+class ConnectionType {
+  static readonly none = 0
+  static readonly left = 1 << 0
+  static readonly right = 1 << 1
+  static readonly top = 1 << 2
+  static readonly bottom = 1 << 4
+}
+
+const makeEyeBit = (ctx: CanvasRenderingContext2D, info: CellInfo, connectionType: number) => {
+  const centerX = info.left + info.size / 2
+  const centerY = info.top + info.size / 2
+  const halfSize = info.size % 2 === 0 ? info.size / 2 : Math.ceil(info.size / 2)
+
+  ctx.fillStyle = info.fillStyle
+
+  const isLeft = (connectionType & ConnectionType.left) === ConnectionType.left
+  const isRight = (connectionType & ConnectionType.right) === ConnectionType.right
+  const isTop = (connectionType & ConnectionType.top) === ConnectionType.top
+  const isBottom = (connectionType & ConnectionType.bottom) === ConnectionType.bottom
+
+  if (isLeft) {
+    ctx.fillRect(centerX, info.top, halfSize, info.size)
+  }
+  if (isRight) {
+    ctx.fillRect(info.left, info.top, halfSize, info.size)
+  }
+  if (isTop) {
+    ctx.fillRect(info.left, info.top, info.size, halfSize)
+  }
+  if (isBottom) {
+    ctx.fillRect(info.left, centerY, info.size, halfSize)
+  }
+  if (((isLeft && !isRight) || (!isLeft && isRight)) && ((isTop && !isBottom) || (!isTop && isBottom))) {
```

Comment:
Is this logically equivalent?

```suggestion
  if (isLeft !== isRight && isTop !== isBottom) {
```